src/com/palmergames/bukkit/towny/war/siegewar/playeractions/InvadeTown.java

```diff
@@ -0,0 +1,187 @@
+				}
+				double distance = Math.sqrt(Math.pow(capitalCoord.getX() - townCoord.getX(), 2) + Math.pow(capitalCoord.getZ() - townCoord.getZ(), 2));
+				if (distance > TownySettings.getNationRequiresProximity()) {
+					throw new TownyException(String.format(TownySettings.getLangString("msg_err_town_not_close_enough_to_nation"), town.getName()));
+				}
+			}
+
+			if (TownySettings.getMaxTownsPerNation() > 0) {
+				if (attackerWinner.getTowns().size() >= TownySettings.getMaxTownsPerNation()){
+					throw new TownyException(String.format(TownySettings.getLangString("msg_err_nation_over_town_limit"), TownySettings.getMaxTownsPerNation()));
+				}
+			}
+
+			captureTown(plugin, siege, attackerWinner, town);
+
+        } catch (TownyException x) {
+            event.setCancelled(true);
+            TownyMessaging.sendErrorMsg(player, x.getMessage());
+        }
+    }
+
+    private static void captureTown(Towny plugin, Siege siege, Nation attackingNation, Town defendingTown) {
+        siege.setTownInvaded(true);
+		
+        //Reset revolt immunity, to prevent immediate revolt after invasion 
+        SiegeWarTimeUtil.activateRevoltImmunityTimer(defendingTown);
+		
+        if(defendingTown.hasNation()) {
+            Nation nationOfDefendingTown = null;
+            try {
+                nationOfDefendingTown = defendingTown.getNation();
+            } catch (NotRegisteredException x) {
+            }
+
+            //Remove town from nation (and nation itself if empty)
+            removeTownFromNation(plugin, defendingTown, nationOfDefendingTown);
+
+            addTownToNation(plugin, defendingTown, attackingNation);
+
+            TownyMessaging.sendGlobalMessage(ChatTools.color(String.format(
+                    TownySettings.getLangString("msg_siege_war_nation_town_captured"),
+                    TownyFormatter.getFormattedTownName(defendingTown),
+                    TownyFormatter.getFormattedNationName(nationOfDefendingTown),
+                    TownyFormatter.getFormattedNationName(attackingNation)
+            )));
+
+            if(nationOfDefendingTown.getTowns().size() == 0) {
+                TownyMessaging.sendGlobalMessage(ChatTools.color(String.format(
+                        TownySettings.getLangString("msg_siege_war_nation_defeated"),
+                        TownyFormatter.getFormattedNationName(nationOfDefendingTown)
+                )));
+            }
+        } else {
+            addTownToNation(plugin, defendingTown, attackingNation);
+
+            TownyMessaging.sendGlobalMessage(ChatTools.color(String.format(
+                    TownySettings.getLangString("msg_siege_war_neutral_town_captured"),
+                    TownyFormatter.getFormattedTownName(defendingTown),
+                    TownyFormatter.getFormattedNationName(attackingNation)
+            )));
+        }
+    }
+
+    private static void removeTownFromNation(Towny plugin, Town town, Nation nation) {
```

Comment:
The following two methods removeTownFromNation and addTownToNation should probably be dropped out for being duplicates of the code in the TownyDatabaseHandler. If changes are made down the road to how towns are added/removed from nations these methods could stop working.

Reply:
Actually scrap that, the only stuff that I don't think is necessary is this:
```
        /*
         * Remove all resident titles/nationRanks before saving the town itself.
         */
        List<Resident> titleRemove = new ArrayList<Resident>(town.getResidents());
		TownyUniverse universe = TownyUniverse.getInstance();

        for (Resident res : titleRemove) {
            if (res.hasTitle() || res.hasSurname()) {
                res.setTitle("");
                res.setSurname("");
            }
            res.updatePermsForNationRemoval(); // Clears the nationRanks.
			universe.getDataSource().saveResident(res);
        }
```

Reply:
Actually scratch that again, I see why it's here and not happening in the Nation object.
I'll move that now and then this will already be handled.

Reply:
I've moved the titleRemove stuff to the Nation object, so you can nuke what i'd quoted in [code] above.

Reply:
- Indicated section nuked